sympy/ntheory/factor_.py

```diff
@@ -2129,3 +2129,59 @@ def is_mersenne_prime(n):
 
     r, b = integer_log(n + 1, 2)
     return b and r in MERSENNE_PRIME_EXPONENTS
+
+
+def _divisors_sum(n):
+    facs = factorint(n)
+    s = 1
+    for k, v in facs.items():
+        s *= (k ** (v + 1) - 1) // (k - 1)
+    return s
+
+
+def is_abundant(n):
+    """Returns True if ``n`` is a abundant number, else False.
+
+    A abundant number is smaller to the sum of its positive, proper divisors.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.factor_ import is_abundant
+    >>> is_abundant(20)
+    True
+    >>> is_abundant(15)
+    False
+
+    References
+    ==========
+
+    .. [1] http://mathworld.wolfram.com/AbundantNumber.html
+
+    """
+    n = as_int(n)
+    return n != 6 and (n % 6 == 0 or _divisors_sum(n) > 2 * n)
+
+
+def is_deficient(n):
+    """Returns True if ``n`` is a deficient number, else False.
+
+    A deficient number is greater to the sum of its positive, proper divisors.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.factor_ import is_deficient
+    >>> is_deficient(20)
+    False
+    >>> is_deficient(15)
+    True
+
+    References
+    ==========
+
+    .. [1] http://mathworld.wolfram.com/DeficientNumber.html
+
+    """
+    n = as_int(n)
+    return _divisors_sum(n) < 2 * n
```

Comment:
I wonder if it would be better to do something like:
```python
def is_deficient(n):
    if isprime(n):
       return True
    if is_perfect(n):
        return False
    return all(is_deficient(d) for d in divisors(n) if 1 < d < n)
```
This is something you could test over a range of numbers to see how it compares.

Reply:
I will compare it and will tell you the results.

Reply:
Also it will give `True` when n is `20`
As 10,5,4,2 all are deficient but 20 is abundant. If a number is deficient, then its divisors has to be deficient but vice-versa is not true.

Reply:
OK. 

What about including the `isprime` and `is_perfect` which can answer without having to factor the number? These can be included
```python
def is_deficient(n):
    if n%2 and is_square(n):
        return True
    if isprime(n):
       return True
    if is_perfect(n):
        return False
    return _divisor_sum(n)<2*n

def is_abundant(n):
    if n!=6 and n%6 == 0:
        return True
    if isprime(n) or is_perfect(n):
       return False
    return _divisor_sum(n) > 2*n
```

Reply:
`isprime` will also check whether this number is prime or not where as `factorint` will return list of 1 and itself. I don`t think that it will help us in optimising the run time as factorint has been already implemented in optimised manner.
We can definitely check is_perfect, Except is_perfect, I don`t think that any check will help us.
We can check the multiples of 6,20,42,56,66 for abundant and also for deficient.

Reply:
Since a multiple of a perfect or abundant number is abundant, it seems like we should be able to handle the abundant case quicker:
```python
def is_abundant(n):
    if n != 6 and n%6 == 0:
        return True
    if isprime(n) or is_perfect(n):
        return False
    facs = factorint(n)
    for k, v in facs.items():
        if v != 1:
            if (k**(v + 1) - 1)//(k - 1) >= 2*k:
                return True
            return False
    return divisor_sigma(n) > 2*n
```
Then `is_deficient(n) == not (is_abundant(n) or is_perfect(n))`.